paddle/operators/gru_op.h

```diff
@@ -202,23 +193,26 @@ class GRUGradKernel : public framework::OpKernel<T> {
       }
 
       math::GRUUnitGradFunctor<Place, T>::compute(
-          context.device_context(), gru_value, gru_grad, frame_size,
-          cur_batch_size,
+          dev_ctx, gru_value, gru_grad, frame_size, cur_batch_size,
           math::ActiveType(context.Attr<std::string>("activation")),
           math::ActiveType(context.Attr<std::string>("gate_activation")));
     }
     if (input_grad) {
       input_grad->mutable_data<T>(context.GetPlace());
       math::Batch2LoDTensorFunctor<Place, T> to_seq;
       batch_gate_grad.set_lod(batch_gate->lod());
-      to_seq(context.device_context(), batch_gate_grad, *input_grad);
+      to_seq(dev_ctx, batch_gate_grad, *input_grad);
     }
     if (bias_grad) {
       bias_grad->mutable_data<T>(context.GetPlace());
-      auto d_b = EigenMatrix<T>::From(*bias_grad);
-      auto d_g = EigenMatrix<T>::From(batch_gate_grad);
-      auto place = context.GetEigenDevice<Place>();
-      d_b.device(place) = d_g.sum(Eigen::array<int, 1>({{0}}));
+      int m = static_cast<int>(batch_gate_grad.dims()[0]);
+      int n = static_cast<int>(batch_gate_grad.dims()[1]);
+      Tensor ones;
+      ones.mutable_data<T>({m}, context.GetPlace());
+      math::SetConstant<Place, T> set;
+      set(dev_ctx, &ones, static_cast<T>(1));
+      math::gemv<Place, T>(dev_ctx, true, m, n, 1., batch_gate_grad.data<T>(),
+                           ones.data<T>(), 0., bias_grad->data<T>());
```

Comment:
Since this is used at many places, should we make a similar function as RowwiseAdd to do this?

Reply:
Done. Add a ColwiseSum functor in `paddle/operators/math/math_function.h` to compute the gradient of bias.